transformer_engine/common/include/transformer_engine/transpose.h

```diff
@@ -11,6 +11,7 @@
 #ifndef TRANSFORMER_ENGINE_TRANSPOSE_H_
 #define TRANSFORMER_ENGINE_TRANSPOSE_H_
 
+#include <vector>
```

Comment:
This header should be compatible with C so you should not use vector here - go with C arrays instead please.